lms/djangoapps/courseware/views/index.py

```diff
@@ -106,18 +114,12 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
         self.section_url_name = section
         self.position = position
         self.chapter, self.section = None, None
-        self.course = None
         self.url = request.path
 
         try:
             set_custom_metrics_for_course_key(self.course_key)
             self._clean_position()
             with modulestore().bulk_operations(self.course_key):
```

Comment:
Well, the only effect of the changes in this file is restricting courseware access to the case where `course_visibility ` is set to `public`. If you want to keep this change as part of this PR, here are a few points:
1. The `get_course_with_access()` needs to be inside a `try` statement as it was originally.
2. I am not sure if there is any performance impact of splitting the work across two `bulk_operations` context managers.

How about instead of moving `get_course_with_access()` just move the access check and redirect to before `self.is_staff = has_access(request.user, 'staff', self.course)` below? That should get the expected result but the course content loading/rendering logic remains exactly the same.

Reply:
Ok, that's a good solution.